File: deliveryRoute/src/main/java/com/sparta/deliveryroute/infrastructure/redis/RedisService.java
package com.sparta.deliveryroute.infrastructure.redis;

import com.sparta.deliveryroute.application.dto.RouteResult;
import lombok.RequiredArgsConstructor;
import org.springframework.data.redis.core.RedisTemplate;
import org.springframework.stereotype.Service;

import java.util.Objects;
import java.util.UUID;

@Service
@RequiredArgsConstructor
public class RedisService {

	private static final String PREFIX = "hut-route::";
	private static final String ARROW = " -> ";
	private final RedisTemplate<String, Object> redisTemplate;

	public RouteResult getHash(String startHubName, String endHubName) {
		return (RouteResult) redisTemplate.opsForHash().get(startHubName, endHubName);
	}

	public UUID getValue(String startHubName, String endHubName) {
		return UUID.fromString(
				Objects.requireNonNull(redisTemplate.opsForValue().get(PREFIX + startHubName + ARROW + endHubName)).toString()
		);
	}
}

File: hub/src/main/kotlin/com/sparta/hub/infrastructure/redis/RedisService.kt
package com.sparta.hub.infrastructure.redis

import com.sparta.hub.application.dto.RouteResult
import org.springframework.data.redis.core.RedisTemplate
import org.springframework.stereotype.Service
import java.util.*

@Service
class RedisService(
    private val redisTemplate: RedisTemplate<String, Any>
) {

    fun setHubRoute(key: String, hubRouteId: UUID) {
        redisTemplate.opsForValue().set(key, hubRouteId)
    }

    fun setHubRouteResult(key1: String, key2: String, value: RouteResult) {
        redisTemplate.opsForHash<String, RouteResult>().put(
            key1,
            key2,
            value
        )
    }

    fun getHubRouteResult(key1: String, key2: String): RouteResult? {
        return redisTemplate.opsForHash<String, RouteResult>().get(key1, key2)
    }

    fun getHubRouteByStartingWithKey(key: String): List<RouteResult> {
        return redisTemplate.opsForHash<String, RouteResult>().values(key)
    }


}